src/CommonDBTM.php

```diff
@@ -2024,6 +2024,10 @@ public function post_updateItem($history = true)
      */
     public function pre_addInDB()
     {
+        if (isset($this->input['name'])) {
```

Comment:
Some classes are not using the default `name` field. You have to use the `getNameField()` to ensure that you are using the correct field.

This remark applies to all your PR.

```suggestion
        if (isset($this->input[static::getNameField()])) {
```